python_modules/libraries/dagster-fivetran/dagster_fivetran/v2/resources.py

```diff
@@ -0,0 +1,155 @@
+import logging
+from enum import Enum
+from typing import Any, Mapping, Optional, Sequence
+
+from dagster import get_dagster_logger
+from dagster._annotations import experimental
+from dagster._config.pythonic_config import ConfigurableResource
+from dagster._record import record
+from dagster._utils.cached_method import cached_method
+from pydantic import Field, PrivateAttr
+from requests.auth import HTTPBasicAuth
+
+
+class FivetranContentType(Enum):
+    """Enum representing each object in Fivetran's ontology."""
+
+    CONNECTOR = "connector"
+    DESTINATION = "destination"
+
+
+@record
+class FivetranContentData:
+    """A record representing a piece of content in a Fivetran workspace.
+    Includes the object's type and data as returned from the API.
+    """
+
+    content_type: FivetranContentType
+    properties: Mapping[str, Any]
+
+    def to_cached_data(self) -> Mapping[str, Any]:
+        return {"content_type": self.content_type.value, "properties": self.properties}
+
+    @classmethod
+    def from_cached_data(cls, data: Mapping[Any, Any]) -> "FivetranContentData":
```

Comment:
why is this one necessary? Feel like this class should just be `@whitelist_for_serdes` if we need to cache.

Reply:
Thanks for the feedback! 

This was the [original pattern](https://github.com/dagster-io/dagster/pull/23486) for Power BI, but I see that [it was updated](https://github.com/dagster-io/dagster/pull/24403) to use `@whitelist_for_serdes` since then. 

I used based my pattern here on the Tableau integration, which was not updated to use `@whitelist_for_serdes` yet.

Changes for this PR made in [4dfe34d](https://github.com/dagster-io/dagster/pull/25750/commits/4dfe34d042baf8f6777d822330585560715cbbbd), I will also update Tableau in another PR.